app/models/project.rb

```diff
@@ -50,6 +51,11 @@ def budget_status
     budget - hours.sum(:value) if budget
   end
 
+  def update_user_activity
+    user_activity = DateTime.now
+    save
+  end
```

Comment:
I would say that updating an old existing entry does indicate activity on the project, and also this is something that is a relatively rare event so adding these callbacks doesn't really seem worth the hassle/extra code to me :)